source/type/data.cpp

```diff
@@ -489,26 +485,62 @@ void Data::aggregate_odt() {
     }
 }
 
-void Data::fill_stop_point_address() {
+void Data::fill_stop_point_address(
+    const std::unordered_map<std::string, navitia::type::Address*>& address_by_address_id) {
     idx_t without_address = 0;
+    size_t addresses_from_ntfs_counter = 0;
+    bool ntfs_addresses_allowed = false;
+    if (!address_by_address_id.empty()) {
+        ntfs_addresses_allowed = true;
+        LOG4CPLUS_INFO(log4cplus::Logger::getInstance("logger"),
+                       "addresses from ntfs are present: " << address_by_address_id.size() << " nb addresses");
+    }
     for (auto sp : pt_data->stop_points) {
         if (!sp->coord.is_initialized()) {
             ++without_address;
             continue;
         }
-        try {
-            auto nb_way = geo_ref->nearest_addr(sp->coord);
-            sp->address = new navitia::georef::Address(nb_way.second, sp->coord, nb_way.first);
-        } catch (const navitia::proximitylist::NotFound&) {
-            LOG4CPLUS_DEBUG(log4cplus::Logger::getInstance("logger"),
-                            "unable to find a way from coord [" << sp->coord.lon() << "-" << sp->coord.lat() << "]");
-            ++without_address;
+        // parse address from ntfs files
+        if (ntfs_addresses_allowed && !sp->address_id.empty()) {
+            auto it = address_by_address_id.find(sp->address_id);
+            if (it != address_by_address_id.end()) {
+                // convert house number string into an int.
+                // We need to clean the field, because it is possible
+                // to have "FACE AU 12"
+                std::string hn_without_information =
+                    std::regex_replace(it->second->house_number, std::regex("[^0-9]*([0-9]+).*"), std::string("$1"));
+                int hn_int = std::stoi(hn_without_information);
+
```

Comment:
``` std::atoi ``` for empty string will give 0 , No ?

Reply:
Yes. This is areal question.

_What do we do with empty number and a number = zero ?_
Because the fied number has to be fill with something (like a 0 for an empty number)
Maybe the question concern more the serialize to rendering this result into the json response, https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py#L347

For the moment, we expose the number even if it equals to 0. It can be a possibility to say : _Hey it is an empty field._ 
What do you think about that ? @kadhikari @azime